cfme/containers/node.py

```diff
@@ -2,31 +2,102 @@
 from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import CheckboxTable, toolbar as tb, paginator, InfoBlock, match_location
+from cfme.web_ui import CheckboxTable, toolbar as tb, InfoBlock, match_location
 from utils.appliance import Navigatable
-from utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
-from cfme.containers.provider import pol_btn, mon_btn
+from utils.appliance.implementations.ui import CFMENavigateStep, navigator
+from utils.log import logger
 
 list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table")
 
 match_page = partial(match_location, controller='container_node', title='Nodes')
 
+# TODO Replace with resource table widget
+resource_locator = "//div[@id='records_div']/table//span[@title='{}']"
+
+
+class NodeView(BaseLoggedInPage):
+    title = Text('#explorer_title_text')
+
+    monitor = Dropdown('Monitoring')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
+
+    nodes = Table(locator="//div[@id='list_grid']//table")
+
+    @property
+    def in_cloud_instance(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Compute', 'Containers', 'Container Nodes'] and
+            match_page()  # No summary, just match controller and title
+        )
+
+
+class NodeCollection(Navigatable):
+    """Collection object for :py:class:`Node`."""
+
+    def instantiate(self, name, provider):
+        return Node(name=name, provider=provider, appliance=self.appliance)
+
+    def all(self):
+        # container_nodes table has ems_id, join with ext_mgmgt_systems on id for provider name
+        node_table = self.appliance.db['container_nodes']
+        ems_table = self.appliance.db['ext_management_systems']
+        node_query = self.appliance.db.session.query(node_table.name, ems_table.name)\
+            .join(ems_table, node_table.ems_id == ems_table.id)
+        nodes = []
+        for name, provider_name in node_query.all():
+            # Hopefully we can get by with just provider name?
+            nodes.append(Node(name=name,
+                              provider=ContainersProvider(name=provider_name,
+                                                          appliance=self.appliance),
+                              collection=self))
+        return nodes
+
+
+class NodeAllView(NodeView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_cloud_instance and
+            match_page(summary='Nodes')
+        )
+
+    paginator = PaginationPane()
+
+
+@navigator.register(NodeCollection, 'All')
+class All(CFMENavigateStep):
+    VIEW = NodeAllView
+    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+
+    def step(self):
+        self.parent_view.navigation.select('Compute', 'Containers', 'Container Nodes')
+
+    def resetter(self):
+        # Reset view and selection
+        tb.select("List View")
```

Comment:
noted, checking that out. Will likely wait until a few of those new widgets are ready, maybe even my paginator #4084